S3/ConnMan.py

```diff
@@ -73,9 +86,20 @@ def _ssl_context():
         cafile = cfg.ca_certs_file
         if cafile == "":
             cafile = None
+        certfile = cfg.ssl_cert_file
+        if certfile == "":
+            certfile = None
+        keyfile = cfg.ssl_key_file
+        if keyfile == "":
+            keyfile = None
+
         debug(u"Using ca_certs_file %s", cafile)
+        debug(u"Using ssl_cert_file %s", certfile)
+        debug(u"Using ssl_key_file %s", keyfile)
 
-        if cfg.check_ssl_certificate:
+        if keyfile is not None and certfile is not None:
```

Comment:
I think that you should modify the condition to:
> if certifle is not None:

If I understand well, it is possible to have just a certifle and not a keyfile (because the private keys are embedded into the certifle), but the opposite is not true.
And the rest of the code should be still ok even if "keyfile" value is None.

Reply:
Yeah, you were right!